ApiFirst/AppServices/Services/PortfolioProductAppService.cs

```diff
@@ -0,0 +1,34 @@
+﻿using AppServices.Interfaces;
+using DomainModels.Models;
+using DomainServices.Interfaces;
+using System;
+using System.Threading.Tasks;
+
+namespace AppServices.Services
+{
+    public class PortfolioProductAppService : IPortfolioProductAppService
+    {
+        private readonly IPortfolioProductService _portfolioProductService;
+
+        public PortfolioProductAppService(IPortfolioProductService portfolioProductServices)
+        {
+            _portfolioProductService = portfolioProductServices ??
+                throw new ArgumentNullException(nameof(portfolioProductServices));
+        }
+
+        public async Task InitRelationAsync(Portfolio portfolio, Product product)
+        {
+            await _portfolioProductService.InitRelationAsync(portfolio, product);
```

Comment:
E nos outros métodos...
```suggestion
            await _portfolioProductService.InitRelationAsync(portfolio, product).ConfigureAwait(false);
```